src/main/kotlin/kweb/Kweb.kt

```diff
@@ -271,7 +280,18 @@ class Kweb private constructor(
         }
     }
 
-    private fun Routing.createHtmlDocumentSupplier(): () -> Document {
+    suspend fun respondKweb(pipelineContext: ApplicationCall, buildPage: WebBrowser.() -> Unit) {
+        val htmlDocument = createHtmlDocumentSupplier().invoke()
+
+        // It's important to use a clone of the template because the result will be modified
+        for (plugin in plugins) {
+            applyPluginWithDependencies(plugin = plugin, appliedPlugins = mutableAppliedPlugins, document = htmlDocument.clone(), routeHandler = pipelineContext.application.routing {  })
```

Comment:
The way the plugins are applied works and is great for the standard use case, but is pretty basic. Ideally we'd want to do a DFS traversal of the plugin tree. I think this would be a great issue for some CS undergrad!